Yosemite/Yosemite/Tools/POS/PointOfSaleOrderService.swift

```diff
@@ -0,0 +1,219 @@
+import Foundation
+import Networking
+
+public struct PointOfSaleCartProduct {
+    let productID: Int64
+    let price: String
+    let productType: ProductType
+
+    public init(productID: Int64, price: String, productType: ProductType) {
+        self.productID = productID
+        self.price = price
+        self.productType = productType
+    }
+}
+
+public struct PointOfSaleCartItem {
+    /// Nil when the cart item is local and has not been synced remotely.
+    let itemID: Int64?
+    let product: PointOfSaleCartProduct
+    let quantity: Decimal
+
+    public init(itemID: Int64?, product: PointOfSaleCartProduct, quantity: Decimal) {
+        self.itemID = itemID
+        self.product = product
+        self.quantity = quantity
+    }
+}
+
+public struct PointOfSaleOrder {
+    public let siteID: Int64
+    public let orderID: Int64
+    public let total: String
+    public let totalTax: String
+    public let currency: String
+    let items: [PointOfSaleOrderItem]
+
+    static func createFromOrder(_ order: Order) -> PointOfSaleOrder {
```

Comment:
nit: could also be an initializer

```suggestion
    init(order: Order) {
```